From 40a254a83dba8bf3b9030fb35b6d4e00eca46e01 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@ubuntu.com>
Date: Wed, 14 Apr 2010 10:00:18 +0100
Subject: [PATCH] dpkg: fix rename deferral

Modern tar files typically use NormalFile1 rather than NormalFile0 for
file objects.  A typo meant that the former never triggered rename
deferral.
---
 debian/changelog |    5 +++++
 src/archives.c   |    2 +-
 2 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/debian/changelog b/debian/changelog
index 3c14cc8..b385b80 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -27,6 +27,11 @@ dpkg (1.15.6.2) UNRELEASED; urgency=low
     should now start using dpkg-buildflags instead. Closes: #560070
   * For Ubuntu set default value of LDFLAGS to -Wl,-Bsymbolic-functions.
 
+  [ Colin Watson ]
+  * Modern tar files typically use NormalFile1 rather than NormalFile0 for
+    file objects.  A typo meant that the former never triggered rename
+    deferral.
+
   [ Updated dpkg translations ]
   * German (Sven Joachim).
 
diff --git a/src/archives.c b/src/archives.c
index 79fef4e..0608137 100644
--- a/src/archives.c
+++ b/src/archives.c
@@ -779,7 +779,7 @@ int tarobject(struct TarInfo *ti) {
    * in dpkg-new.
    */
 
-  if (ti->Type == NormalFile0 || ti->Type == NormalFile0) {
+  if (ti->Type == NormalFile0 || ti->Type == NormalFile1) {
     nifd->namenode->flags |= fnnf_deferred_rename;
 
     debug(dbg_eachfiledetail, "tarobject done and installation deferred");
-- 
1.7.0

